pkg/util/taints/taints.go

```diff
@@ -205,9 +206,9 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 	return allErrs, removed
 }
 
-// addTaints adds the newTaints list to existing ones and updates the newTaints List.
-// TODO: This needs a rewrite to take only the new values instead of appended newTaints list to be consistent.
-func addTaints(oldTaints []v1.Taint, newTaints *[]v1.Taint) bool {
+// addedTaints if there are taints in old but not in new. Would return the taints. But only return true when len(oldTains) != (len(*newTaints) + len(addedTaints))
+// this means that if only there are taints in the newTaints but not the oldTaints, return nil and true.
+func addedTaints(oldTaints []v1.Taint, newTaints *[]v1.Taint) (addedTaints []v1.Taint, added bool) {
```

Comment:
I like the method rename. Original name is confusing next to `deleteTaints`. Not sure what the move of append from this method to the callee gives though.  Do you have an idea?

Reply:
How do you think of addableTaints and toAddedTaints?